src/battle/overlay_12_0224E4FC.c

```diff
@@ -1592,3 +1593,180 @@ int ov12_02250CFC(BATTLECONTEXT *ctx, int battlerId) {
+    int trainerIndex;
+    
+    if (state & 0x84) {
+        return FALSE;
+    }
+    
+    if (!(state & BATTLE_TYPE_TRAINER)) {
+        return FALSE;
+    }
+    
+    if (state & BATTLE_TYPE_DOUBLES) {
+        return FALSE;
+    }
+    
+    trainerIndex = BattleSys_GetTrainerIndex(bsys, 1);
+    state = 0;
+    
+    do {
+        switch (state) {
+        case 0:
+            if (ctx->battleMons[1].unk78 == 1 && !(ctx->linkStatus2 & 0x20) && TrainerMessageWithIdPairExists(trainerIndex, 13, HEAP_ID_BATTLE)) {
+                ctx->linkStatus2 |= 0x20;
+                ctx->msgWork = 13;
+                return TRUE;
+            }
+            state++;
+            break;
+        case 1:
+            if (!(ctx->battleMons[1].msgFlag & 2) && ctx->battleMons[1].hp <= ctx->battleMons[1].maxHp / 2 && TrainerMessageWithIdPairExists(trainerIndex, 14, HEAP_ID_BATTLE)) {
+                ctx->battleMons[1].msgFlag |= 2;
+                ctx->msgWork = 14;
+                return TRUE;
+            }
+            state++;
+            break;
+        case 2:
+            if (!(ctx->battleMons[1].msgFlag & 3)) {
+                int i;
+                int aliveMons;
+                PARTY *party;
+                Pokemon *mon;
+                
+                party = BattleSys_GetParty(bsys, 1);
+                aliveMons = 0;
+                
+                for (i = 0; i < GetPartyCount(party); i++) {
+                    mon = GetPartyMonByIndex(party, i);
+                    if (GetMonData(mon, MON_DATA_HP, NULL)) {
+                        aliveMons++;
+                    }
+                }
+                if (aliveMons == 1 && TrainerMessageWithIdPairExists(trainerIndex, 15, HEAP_ID_BATTLE)) {
+                    ctx->battleMons[1].msgFlag |= 3;
+                    ctx->msgWork = 15;
+                    return TRUE;
+                }
+            }
+            state++;
+            break;
+        case 3:
+            if (!(ctx->battleMons[1].msgFlag & 4)) {
+                int i;
+                int aliveMons;
+                PARTY *party;
+                Pokemon *mon;
+                
+                party = BattleSys_GetParty(bsys, 1);
+                aliveMons = 0;
+                
+                for (i = 0; i < GetPartyCount(party); i++) {
+                    mon = GetPartyMonByIndex(party, i);
+                    if (GetMonData(mon, MON_DATA_HP, NULL)) {
+                        aliveMons++;
+                    }
+                }
+                if (aliveMons == 1 && (ctx->battleMons[1].hp <= ctx->battleMons[1].maxHp / 2) && TrainerMessageWithIdPairExists(trainerIndex, 16, HEAP_ID_BATTLE)) {
+                    ctx->battleMons[1].msgFlag |= 4;
+                    ctx->msgWork = 16;
+                    return TRUE;
+                }
+            }
+            state++;
+            break;
+        case 4:
+            break;
+        }
+    } while (state != 4);
```

Comment:
this is likely required for matching, but if not, would a for loop work here?

Reply:
a for loop would be a bit awkward since it'd have to be `for (state = 0; state != 4; )` since that last case 4 doesn't increment the state variable